src/analysis/processing/qgsalgorithmdbscanclustering.cpp

```diff
@@ -133,11 +139,13 @@ QVariantMap QgsDbscanClusteringAlgorithm::processAlgorithm( const QVariantMap &p
   feedback->pushInfo( QObject::tr( "Analysing clusters" ) );
   std::unordered_map< QgsFeatureId, int> idToCluster;
   idToCluster.reserve( index.size() );
+  std::unordered_map< int, int> clusterSize;
   QgsFeatureIterator features = source->getFeatures( QgsFeatureRequest().setNoAttributes() );
   const long featureCount = source->featureCount();
+  dbscan( minSize, eps, borderPointsAreNoise, featureCount, features, index, idToCluster, clusterSize, feedback );
```

Comment:
passing clusterSize here shouldn't be required -- dbscan doesn't do anything useful with it

Reply:
I'm passing it on so I can initiate the map with all cluster IDs and a count of 0. That allows for a nice one line loop to compute the  cluster size afterwards. You prefer if I'd go at it another way?

Reply:
It shouldn't be needed -- the call to ` clusterSize[ idCluster.second ]` would automatically add the missing id anyway and initialize it to 0.

Reply:
TIL moment, thanks